### PR TITLE
More single phase outputs

### PR DESCRIPTION
**Pull request by MelReyCG**
The point of this PR is to add to the statistics :
- The region temperature (min, average, max),
- The fluid mass in the region.

Exemple of the new statistics block :
```
compflowStatistics, Reservoir: Pressure (min, average, max): -4.23826e+08, -3.3097e+07, 3.18951e+07 Pa
compflowStatistics, Reservoir: Delta pressure (min, max): -4.43974e+08, 0 Pa
compflowStatistics, Reservoir: Temperature (min, average, max): 366.483, 366.483, 366.483 K
compflowStatistics, Reservoir: Total dynamic pore volume: 5.272e+11 rm^3
compflowStatistics, Reservoir: Total fluid mass: 5.90061e+14 kg
```
(new lines are `Temperature` & `Total fluid mass`)

### CODE REVIEW

File: src/coreComponents/physicsSolvers/fluidFlow/SinglePhaseBaseKernels.hpp [hunk 1]
```diff
@@ -674,15 +674,21 @@ struct StatisticsKernel
           arrayView1d< real64 const > const & volume,
           arrayView1d< real64 const > const & pres,
           arrayView1d< real64 const > const & deltaPres,
+          arrayView1d< real64 const > const & temp,
           arrayView1d< real64 const > const & refPorosity,
           arrayView2d< real64 const > const & porosity,
+          arrayView2d< real64 const > const & densities,
```
Comment (paveltomin) [inline src/coreComponents/physicsSolvers/fluidFlow/SinglePhaseBaseKernels.hpp]: can you name it `density` for consistency?

File: src/coreComponents/physicsSolvers/fluidFlow/SinglePhaseStatistics.cpp [hunk 2]
```diff
@@ -176,21 +206,32 @@ void SinglePhaseStatistics::computeRegionStatistics( MeshLevel & mesh,
...
-      regionStatistics.averagePressure /= regionStatistics.totalUncompactedPoreVolume;
+      float invTotalUncompactedPoreVolume = 1.0 / regionStatistics.totalUncompactedPoreVolume;
+      regionStatistics.averagePressure *= invTotalUncompactedPoreVolume;
+      regionStatistics.averageTemperature *= invTotalUncompactedPoreVolume;
     }
     else
     {
       regionStatistics.averagePressure = 0.0;
+      regionStatistics.averageTemperature = 0.0;
       GEOS_LOG_LEVEL_RANK_0( 1, getName() << ", " << regionNames[i]
```
Comment (paveltomin) [inline src/coreComponents/physicsSolvers/fluidFlow/SinglePhaseStatistics.cpp]: should it be warning?

**Review by paveltomin — APPROVED** [review]

File: src/coreComponents/physicsSolvers/fluidFlow/SinglePhaseStatistics.cpp (hunk 2 above)
Comment (MelReyCG) [reply src/coreComponents/physicsSolvers/fluidFlow/SinglePhaseStatistics.cpp]: Indeed, as those values can be silently output in a `.csv`, we should output that as a warning.

**Review by dkachuma — APPROVED** [review]
This looks good to me.
Do we also need to do this for `CompositionalMultiphaseStatistics` or is it already done?

**Comment by MelReyCG** [comment]
@dkachuma Thanks for your feedback.
It is indeed already done for `CompositionalMultiphaseStatistics`.

**Comment by MelReyCG** [comment]
I think we are ready to merge.
```
ATS END OF RUN STARTED January 15, 2024 13:42:41
    Status     : TestCase                              : Directory                                                 : Elapsed : Resources : TestStep  
    ---------- : ------------------------------------- : --------------------------------------------------------- : ------- : --------- : ----------
    FAIL CHECK : PoroElastic_impermeableFault_smoke_01 : poroElasticCoupling/PoroElastic_impermeableFault_smoke_01 : 0:00:05 : 0:00:05   : restartcheck
    FAIL CHECK : PoroElastic_permeableFault_smoke_01   : poroElasticCoupling/PoroElastic_permeableFault_smoke_01   : 0:00:05 : 0:00:05   : restartcheck
    ---------- : ------------------------------------- : --------------------------------------------------------- : ------- : --------- : ----------
```

**Pull request merged**